### ISSUE TITLE
Make Gibbs kernels composable

### ISSUE THREAD

**Issue by fehiepsi** [pyro-ppl/numpyro#898]
Currently, our design in HMCGibbs, DiscreteHMCGibbs, HMCECS does not allow to compose those kernels. However, I believe that with just a few internal changes, we should be able to compose those kernels to get more flexible kernels.

### Proposal

TODO

**Comment by fehiepsi** [comment]
@juanitorduz @Qazalbash Do you think we can redesign MCMC kernels to make them composable with the help of claude code? The idea was to apply each kernel for just a subset of variables and MCMC runs still work. Curious on what do you think?

**Comment by juanitorduz** [comment]
> [@juanitorduz](https://github.com/juanitorduz) [@Qazalbash](https://github.com/Qazalbash) Do you think we can redesign MCMC kernels to make them composable with the help of claude code? The idea was to apply each kernel for just a subset of variables and MCMC runs still work. Curious on what do you think?

Yes! Makes a lot of sense! I will create a PR with the scaffolding design (not the code itself), and then we can proceed with the implementation :) . I have some tokens I can burn for this task :)
